### ISSUE TITLE
About Lipschitz Constant Regularization

### ISSUE THREAD

**Issue by Zengyf-CVer** [BCV-Uniandes/Gabor_Layers_for_Robustness#2]
@juancprzs 
I studied your paper and code carefully. I have a question. Which part of the code in `gabor_layers.py` reflects Lipschitz Constant Regularization?
![222](https://user-images.githubusercontent.com/41098760/141431293-4e031c4c-c15d-47fe-a25d-15701fd636c6.PNG)

**Comment by juancprzs** [comment]
Hi there,

Thank you for your interest in our paper!

As we mention in Section *3.4 Lipschitz Constant Regularization*, based on the expression we find for the Lipschitz constant, we regularize \sigma directly. 

Implementation wise, in the code [this](https://github.com/BCV-Uniandes/Gabor_Layers_for_Robustness/blob/master/main.py#L83) parameter controls the regularization term. In turn, the regularization term goes into the loss in [this](https://github.com/BCV-Uniandes/Gabor_Layers_for_Robustness/blob/master/main.py#L539) line.

Please let me know if you need further information.

**Comment by Zengyf-CVer** [comment]
@juancprzs 
In other words, the `Lipschitz Constant Regularization` in the paper is not in GaborLayer, but is used for loss calculation in training?

**Comment by juancprzs** [comment]
Exactly. Regularization is part of the training of the layer, not of the layer itself. 

Of course, in essence, the fact that the regularizer takes this form is a direct consequence of the layer being a Gabor Layer of the form we use.

**Comment by Zengyf-CVer** [comment]
@juancprzs 
Somewhat similar to Conv+BN?

**Comment by Suvashsharma** [comment]
Hey there,
In the paper, as per equation (3), the regularization loss proposed is bounded using tanh of sigma. However, I see the code uses the direct minimization of sigma^2. Is that an update yet to come or did I misunderstand your code?

**Comment by juancprzs** [comment]
Hey @Suvashsharma. Thanks for your interest!

Indeed, the code branch we published of our implementation corresponds to Equation (2), not Equation (3). This is because, for most setups, we found this alternative to be more useful. If one wants to use Equation (3), the modification in the code amounts simply to adding the `tanh` function and, if you want, the `\mu` regularization term.

**Comment by Lisa1999** [comment]
What is the right value for this
